pkg/dynatrace/dynatrace.go

```diff
@@ -247,48 +268,51 @@ func (dynatraceApiClient *DynatraceApiClient) CreateDynatraceHttpMonitor(monitor
 	return monitorId, nil
 }
 
-func (dynatraceApiClient *DynatraceApiClient) ExistsHttpMonitorInDynatrace(monitorId string) (bool, error) {
-	path := ("/synthetic/monitors/")
-	resp, err := dynatraceApiClient.MakeRequest(http.MethodGet, path, "")
+func (dynatraceApiClient *DynatraceApiClient) ExistsHttpMonitorInDynatrace(clusterId string) (bool, error) {
+	existsHttpMonitorResponse, err := dynatraceApiClient.GetDynatraceHttpMonitors(clusterId)
 	if err != nil {
 		return false, err
 	}
-	defer resp.Body.Close()
 
-	// Check if the response status code is OK
-	if resp.StatusCode != http.StatusOK {
-		return false, fmt.Errorf("failed to fetch monitor in Dynatrace. Status code: %d", resp.StatusCode)
-	}
+	countMonitors := len(existsHttpMonitorResponse.Monitors)
 
-	var existsHttpMonitorResponse ExistsHttpMonitorInDynatraceResponse
-	if err := json.NewDecoder(resp.Body).Decode(&existsHttpMonitorResponse); err != nil {
-		return false, fmt.Errorf("error parsing JSON: %w", err)
-	}
+	switch {
+	case countMonitors == 1:
+		return true, nil
 
-	for _, monitor := range existsHttpMonitorResponse.Monitors {
-		if monitor.EntityId == monitorId {
-			return true, nil
+	case countMonitors == 0:
+		return false, nil
+
+	case countMonitors > 1:
+		for i := 1; i < countMonitors; i++ {
+			if err := dynatraceApiClient.DeleteDynatraceMonitorByCluserId(clusterId); err != nil {
+				return false, fmt.Errorf("failed to delete monitors for cluster id %s: %w", clusterId, err)
+			}
 		}
```

Comment:
Would it make more sense to split the deletion logic out of this function?

I personally wouldn't expect calling a function named `ExistsHTTPMonitorInDynatrace()` to have the side-effect of deleting extraneous monitors. If there's a technical reason why we must include it in this function, can we be sure to document it with comments?